tests/test_browser.py

```diff
@@ -0,0 +1,108 @@
+from typing import NoReturn
+from pytest_webdriver_recorder.browser import Chrome, BrowserError
+import pytest
+import time
+
+
+@pytest.fixture
+def url(local_html_path):
+    return f'file://{local_html_path}'
+
+
+@pytest.fixture(scope='session')
+def session_browser():
+    """Instantiating this is slow; to speed up tests, we'll use a single instance, but clean it up before each test."""
+    return Chrome()
+
+
+@pytest.fixture
+def browser(session_browser, url) -> Chrome:
+    session_browser.pngs = []
+    with session_browser.autocapture_off():
+        session_browser.get(url)
+    return session_browser
+
+
+def be_boundless_and_wait(browser: Chrome) -> NoReturn:
+    browser.send_inputs('be boundless')
+    browser.click_button('update')
+    browser.wait_for('p', 'be boundless')
+
+
+def test_context_stops_client_on_exit(url):
+    class TestChrome(Chrome):
+        is_stopped = False
+
+        def stop_client(self):
+            self.is_stopped = True
+
+    with TestChrome() as b:
+        b.get(url)
+
+    assert b.is_stopped
+
+
+def test_fill_and_clear(browser):
+    browser.send_inputs('boundless')
+    assert browser.switch_to.active_element.get_attribute('value') == 'boundless'
+    browser.clear()
+    assert not browser.switch_to.active_element.get_attribute('value')
+
+
+def test_run_commands(browser):
+    browser.run_commands([
+        ('send_inputs', 'boundless'),
+        ('click_button', 'update'),
+        ('wait_for', 'p', 'boundless')
+    ])
+
+
+def test_snap(browser):
+    assert not browser.pngs
+    browser.snap()
+    assert len(browser.pngs) == 1
+
+
+def test_send(browser):
+    time.sleep(.5)
```

Comment:
Need a better mechanism to wait until the page is ready. If it goes too fast the result in the first field winds up as `oo` or `o` instead of `foo`, which I confirmed with screenshots also.